### ISSUE TITLE
Finetune Training: The process cannot access the file because it is being used by another process

### ISSUE THREAD

**Issue by gamekiller2048** [erew123/alltalk_tts#563]
When running the finetune training step, I get the error:

[FINETUNE] ERROR: [INFO] An unexpected error occurred: [WinError 32] The process cannot access the file because it is being used by another process: 'D:/alltalk_tts/finetune/Jumper/training/XTTS_FT-March-16-2025_09+27PM-9317203\\trainer_0_log.txt'

I've tried:
- checking file locks and only the python process has a lock on the file when running
- restarting with different projects/datasets
- similar to this discussion: https://github.com/erew123/alltalk_tts/discussions/340

This is the cmd log
`Running on local URL:  http://0.0.0.0:7052
[FINETUNE] HTTP Request: GET http://localhost:7052/startup-events "HTTP/1.1 200 OK"
[FINETUNE] HTTP Request: HEAD http://localhost:7052/ "HTTP/1.1 200 OK"

To create a public link, set `share=True` in `launch()`.
[FINETUNE] [INFO] Initializing output directory: D:\alltalk_tts\finetune\Jumper
[FINETUNE] [MODEL] Using device: cuda
[FINETUNE] [MODEL] Loading Whisper model: tiny
[FINETUNE] [MODEL] Using mixed precision
[FINETUNE] [MODEL] Initializing Silero VAD
Using cache found in D:\.cache\torch\hub\snakers4_silero-vad_master
[FINETUNE] [INFO] Updated language to: en
[FINETUNE] [INFO] Processing: Jumper-[AudioTrimmer.com]
[FINETUNE] [INFO] Finalizing processing
100%|################################################################################| 17/17 [00:13<00:00,  1.24file/s]
[FINETUNE] [INFO] Total mismatches found: 3
[FINETUNE] [MODEL] ✓ All required files found for model: XTTSv2
[FINETUNE] [MODEL] ***********************
[FINETUNE] [MODEL] Training Configuration:
[FINETUNE] [MODEL] ***********************
[FINETUNE] [MODEL] - Language: en
[FINETUNE] [MODEL] - Epochs: 10
[FINETUNE] [MODEL] - Batch Size: 4
[FINETUNE] [MODEL] - Gradient Accumulation: 1
[FINETUNE] [MODEL] - Learning Rate: 5e-06
[FINETUNE] [MODEL] - Learning Rate Scheduler: CosineAnnealingWarmRestarts
[FINETUNE] [MODEL] - Optimizer: AdamW
[FINETUNE] [MODEL] - Number of Workers: 8
[FINETUNE] [MODEL] - Warm Up: False
[FINETUNE] [MODEL] - Max Audio Length: 242550
[FINETUNE] [INFO] **********************
[FINETUNE] [INFO] Project Configuration:
[FINETUNE] [INFO] **********************
[FINETUNE] [INFO] - Project Path: D:\alltalk_tts\finetune\Jumper\training
[FINETUNE] [INFO] - Model Path: D:\alltalk_tts\models\xtts\XTTSv2
[FINETUNE] [INFO] - Training Data: D:\alltalk_tts\finetune\Jumper\metadata_train.csv
[FINETUNE] [INFO] - Evaluation Data: D:\alltalk_tts\finetune\Jumper\metadata_eval.csv
[FINETUNE] [INFO] - Language: en
[FINETUNE] [INFO] - Found language file, using language: en
[FINETUNE] [INFO] - Batch Size: 4
[FINETUNE] [INFO] - Grad Steps: 1
[FINETUNE] [INFO] - Training Epochs: 10
[FINETUNE] [INFO] - Learning Scheduler CosineAnnealingWarmRestarts Parameters
[FINETUNE] [INFO] - {'T_0': 2, 'T_mult': 1, 'eta_min': 1e-06, 'last_epoch': -1}
[FINETUNE] DVAE weights restored from: D:\alltalk_tts\models\xtts\XTTSv2\dvae.pth
[FINETUNE] [MODEL] Loading training samples...
[FINETUNE] Found 15 files in D:\alltalk_tts\finetune\Jumper
[FINETUNE] [MODEL] Loaded 15 training and 2 eval samples

 > Training Environment:
 | > Backend: Torch
 | > Mixed precision: False
 | > Precision: float32
 | > Current device: 0
 | > Num. of GPUs: 1
 | > Num. of CPUs: 8
 | > Num. of Torch Threads: 1
 | > Torch seed: 1
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
 > Start Tensorboard: tensorboard --logdir=D:\alltalk_tts\finetune\Jumper\training\XTTS_FT-March-16-2025_09+27PM-9317203
 > Model has 518442047 parameters

[FINETUNE] [MODEL] ********************************
[FINETUNE] [MODEL] Starting training the XTTS model
[FINETUNE] [MODEL] ********************************

 > EPOCH: 0/10
 --> D:\alltalk_tts\finetune\Jumper\training\XTTS_FT-March-16-2025_09+27PM-9317203
[FINETUNE] Sampling by language: dict_keys(['en'])

 > TRAINING (2025-03-16 21:28:01)
[FINETUNE] ERROR: [INFO] An unexpected error occurred: [WinError 32] The process cannot access the file because it is being used by another process: 'D:/alltalk_tts/finetune/Jumper/training/XTTS_FT-March-16-2025_09+27PM-9317203\\trainer_0_log.txt'`

**Comment by RenNagasaki** [comment]
I always had this problem if my finetuning process crashed before (settings too high or other problems)
IIRC that's not an actual error as in "finetuning stopped" but more that the previous finetuning attempt didn't release the lock on the logfile. It should still be finetuning, but without logging.

I solved this by restarting the finetune script after a crash. 
This should relate to issue https://github.com/erew123/alltalk_tts/issues/275

**Comment by gamekiller2048** [comment]
The error persists even when rerunning the script on different datasets. It seems to happen at the start of the training loop.

Maybe it’s due to my system not having enough resources to even start 1 epoch? The system requirements checker shows all checks except for one.

I think it was 12 G recommended vram but I have like 9. But I’m using the tiny model (not sure if that makes a diff).
Also shouldn’t that throw a Pytorch memory error instead?

**Comment by subspecs** [comment]
Still happens, with 24GB of RAM and a RTX 3090 (24GB VRAM)
And when it gets to that point it barely uses any RAM or VRAM at all, so it's not a resource bottleneck.

**Comment by JoBilert** [comment]
Getting the same error here. RTX3060 with 12 GB VRAM and 128 GB System RAM so RAM should definitely not be an issue.
